nvflare/fuel/f3/cellnet/new_cell.py

```diff
@@ -0,0 +1,155 @@
+            channel=channel, topic=topic, targets=targets, message=message, optional=optional
+        )
+
+    def get_fqcn(self):
+        return self.core_cell.get_fqcn()
+
+    def get_internal_listener_url(self) -> Union[None, str]:
+        """Get the cell's internal listener url.
+
+        This method should only be used for cells that need to have child cells.
+        The url returned is to be passed to child of this cell to create connection
+
+        Returns: url for child cells to connect
+
+        """
+        return self.core_cell.get_internal_listener_url()
+
+    def send_request(self, channel, target, topic, request, timeout=None, optional=False):
+        if channel != CellChannel.SERVER_COMMAND:
+            return self.core_cell.send_request(
+                channel=channel, target=target, topic=topic, request=request, timeout=timeout, optional=optional
+            )
+
+        req_id = str(uuid.uuid4())
+
+        request.add_headers({StreamHeaderKey.STREAM_REQ_ID: req_id})
+        future = self.send_blob(channel, topic, target, request)  # StreamCell API
+
+        # this future can be used to check sending progress, but not for checking return blob
+        waiter = threading.Event()
+        self.requests_dict[req_id] = [waiter, None]
+
+        if not waiter.wait(timeout):
+            self.requests_dict.pop(req_id)
+            return Message()  # or raise TimeoutError(f"request to {channel=} {topic=} timeout")
+        _, result = self.requests_dict.pop(req_id)
+        return result
+
+    def _process_reply(self, future: StreamFuture):
+        req_id = future.headers.get(StreamHeaderKey.STREAM_REQ_ID, -1)
+        if req_id not in self.requests_dict:
+            return False  # response coming back too late.  The req_id was popped out during timeout
+        headers = future.headers
+        response_blob = future.result()
+        waiter, _ = self.requests_dict.get(req_id, [None, None])
+        self.requests_dict[req_id] = [waiter, Message(headers, response_blob)]
+        waiter.set()
+
+    def register_request_cb(self, channel: str, topic: str, cb, *args, **kwargs):
+        """
+        Register a callback for handling request. The CB must follow request_cb_signature.
+
+        Args:
+            channel: the channel of the request
+            topic: topic of the request
+            cb:
+            *args:
+            **kwargs:
+
+        Returns:
+
+        """
+        if not callable(cb):
+            raise ValueError(f"specified request_cb {type(cb)} is not callable")
+        if channel != CellChannel.SERVER_COMMAND:
+            self.core_cell.register_request_cb(channel, topic, cb, *args, **kwargs)
+        else:
+            adapter = Adapter(cb, self.core_cell.my_info, self)
+            self.register_blob_cb(
+                channel, topic, adapter.call, *args, **kwargs
+            )  # this cb is defined by Yuhong.  The cb expects the message, not a future.
```

Comment:
maybe move the comment into method arguments descriptions?

Reply:
Good catch.  Will remove it.